goalgorilla/open_social
# Issue #3224859 by nkoporec: Flexible groups with visibility set to secret, should not show up in views results

## Problem
When I'm logged in as an authenticated user and I go to /all-groups page, I see groups that has the visibility option set to 'Group members only (secret)' even though I'm not part of that group.

## Solution
We need to alter the views query, so that it will only show secret groups that the user is part of.

## Issue tracker
https://www.drupal.org/project/social/issues/3224859

## How to test
1. As a site manager create a group with the visibility set to 'Group members only (secret)'
2. Log in as authenticated user and visit /all-groups page.
3. See that the secret group is displayed, even though I'm not a member of it.

## Screenshots
Before:
![2021-07-22_12-10](https://user-images.githubusercontent.com/35064680/126624465-4f26f856-9436-4ee6-bc9f-ad24489821d2.png)

After:
![2021-07-22_12-08](https://user-images.githubusercontent.com/35064680/126624491-4d98e45d-f1c1-4d60-91b0-0376d9d02c5e.png)


## Release notes
Fixed an issue where groups with visibility set to 'Group members only (secret)' would still show up in the views results.

## Change Record


## Translations
